app/projects/actions.ts

```diff
@@ -1,42 +1,65 @@
 'use server';
 
+import { z } from 'zod';
 import { revalidatePath } from 'next/cache';
-import { type z } from 'zod';
+import { redirect } from 'next/navigation';
 
-import { type schema } from '@/components/ui/project-form';
+import { action } from '@/lib/safe-action';
 import { prisma } from '@/lib/db';
 
-type Project = z.infer<typeof schema>;
+const schema = z.object({
+  id: z.string().optional(),
+  name: z.string().min(1, { message: 'Please enter project name' }),
+  upworkId: z.string().nullable(),
+  hubstaffId: z.string().nullable(),
+  asanaId: z.string().nullable(),
+});
 
-export const addProject = async (data: Project) => {
-  await prisma.project.create({
-    data: {
-      name: data.name,
-      upworkId: data.upworkId,
-      hubstaffId: data.hubstaffId,
-      asanaId: data.asanaId,
-    },
-  });
+export const addProject = action(schema, async (data) => {
+  try {
+    await prisma.project.create({
+      data: {
+        name: data.name,
+        upworkId: data.upworkId,
+        hubstaffId: data.hubstaffId,
+        asanaId: data.asanaId,
+      },
+    });
+  } catch (e) {
+    console.log(e);
 
-  revalidatePath('/projects');
-};
+    return { failure: 'Error occured while adding the project!' };
```

Comment:
Do not return `{ failure }`, just throw! `next-safe-action` will catch is and wrap into error type properly

Reply:
If we don't mind prisma error getting send to the client, try/catch here can be removed completely